### DISCUSSION TITLE
What programs do you recommend writing to get good at Java?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#140607] [Programming Help]
### Body

Hi,

What programs do you recommend writing to get good at Java?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by juacardonahe** [comment]
To get good at Java, try building these types of programs:

To-Do List App – Practice OOP concepts, file handling, and GUIs.
Simple Games – Like Tic-Tac-Toe or Snake, to work on algorithms and logic.
API Integration – Build a weather app or currency converter to practice handling APIs and JSON.
Banking System – Manage accounts, transactions, and user input for practice with data structures.

**Reply by ghost** [reply]
I am thinking of building a movie review system using MySQL, Java, Java Spring and Java SQL library including MySQL.

**Reply by YuheshPandian** [reply]
Thats awesome...That ain't beginner level

**Comment by d-coder111** [comment]
Hey @WorldsDumbestCoder👋😊, to get good at Java, it's essential to practice by building various types of applications that cover different aspects of the language and its ecosystem. Well, I will suggest W3Schools, which offers a comprehensive Java tutorial that is great for beginners. 

### Features
Try it Yourself: Each tutorial comes with a “Try it Yourself” editor, allowing you to edit Java code and see the results instantly.
Examples: Numerous code examples to illustrate concepts clearly.
References: Easy-to-navigate references for Java keywords and functions.
Link-https://www.w3schools.com/java/
You can access the Java tutorial directly here: W3Schools Java Tutorial.

This resource is great for hands-on learning and is suitable for beginners looking to get a solid foundation in Java programming.

Feel free to mark it as the accepted answer if you found it helpful!✌👍

**Comment by VeerammalV** [comment]
To get good at Java, focus on mastering core concepts like object-oriented programming, data structures, algorithms, and concurrency. Dive into Java’s standard libraries and frameworks, experiment with threading and synchronization, and work on optimizing performance through memory management. Practicing problem-solving on platforms like LeetCode or Codeforces can also enhance your skills. Additionally, reading and understanding open-source Java code can expose you to real-world coding practices.

The problems I practiced for learning java : [https://github.com/VeerammalV/KIOT-Shine-Aspire-Systems](url). Just go through it If you want to..
